src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/config/HmppsIncentivesApiExceptionHandler.kt

```diff
@@ -231,6 +245,9 @@ class HmppsIncentivesApiExceptionHandler {
 class NoDataFoundException(id: Long) :
   Exception("No Data found for ID $id")
 
+class NoDataWithCodeFoundException(dataType: String, code: String) :
+  Exception("No $dataType found for code $code")
```

Comment:
I tend to surround these kind of strings in single quote in case there is some accidental whitespace, that can help, e.g. `No IncentiveLevel found for code 'STD '` is less puzzling in case of strange 404s :) (probably more useful when making manual requests to API than in machine-to-machine requests tho)

Reply:
✅